homeassistant/components/sensehat.py

```diff
@@ -0,0 +1,65 @@
+"""
+A component which show the value of temperature, humidity and pressure
+from Sense HAT board in the form of service.
+Interval time between each reading is a minute by default
+but it is configurable from configuration file.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/
+"""
+
+import time
+import os
+from sense_hat import SenseHat
+
+DOMAIN = "sensehat"
+
+
+def get_cpu_temp():
+    # get CPU temperature
+    res = os.popen("vcgencmd measure_temp").readline()
+    t = float(res.replace("temp=", "").replace("'C\n", ""))
+    return t
+
+
+def get_average(x):
+    # use moving average to get better readings
+    if not hasattr(get_average, "t"):
+        get_average.t = [x, x, x]
+    get_average.t[2] = get_average.t[1]
+    get_average.t[1] = get_average.t[0]
+    get_average.t[0] = x
+    xs = (get_average.t[0]+get_average.t[1]+get_average.t[2])/3
+    return xs
+
+sense = SenseHat()
+INTERVAL_DEFAULT = 60
+
+
+def setup(hass, config):
+    # main setup function
+
+    def get_temp(call):
```

Comment:
This is not how a service call works, this method never stops and neither should it be used to update entities. Please remove.